chaintest/action_test_helpers.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+var _ state.Mutable = (*InMemoryStore)(nil)
+
+// InMemoryStore is a storage that acts as a wrapper around a map and implements state.Mutable.
+type InMemoryStore struct {
+	Storage map[string][]byte
+}
+
+func NewInMemoryStore() *InMemoryStore {
+	return &InMemoryStore{
+		Storage: make(map[string][]byte),
+	}
+}
+
+func (i *InMemoryStore) GetValue(_ context.Context, key []byte) ([]byte, error) {
+	val, ok := i.Storage[string(key)]
+	if !ok {
+		return nil, database.ErrNotFound
+	}
+	return val, nil
+}
+
+func (i *InMemoryStore) Insert(_ context.Context, key []byte, value []byte) error {
```

Comment:
What is the expected behaviour on a duplicate key? The interface isn't documented. Similarly, but for a missing key, when calling `Remove()`? Should these return errors?

Reply:
When looking at `Database`, `TStateView`, `DB`, all of these seem to have the same behaviour. In the case of stores that are using a database, they are returning an error if the inner database was closed. For the tstate, if the key does not have enough permissions.
For `Remove`, if the value is `nil` then no error (nil) is returned, which is reflected by this implementation because `delete` won't panic if the value is `nil`.